integrationtests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -348,35 +353,50 @@ private static void assertCreatingTopicThrowsExpectedException(Admin proxyAdmin)
 
     @Test
     void testCompositeFilter() {
-        try (MockServerKroxyliciousTester tester = mockKafkaKroxyliciousTester((mockBootstrap) -> proxy(mockBootstrap)
-                .addToFilters(new FilterDefinitionBuilder(CompositePrefixingFixedClientIdFilter.class.getName())
-                        .withConfig("clientId", "banana", "prefix", "123").build()));
+        final KroxyliciousTesterBuilder testerBuilder = new KroxyliciousTesterBuilder()
```

Comment:
@k-wall @robobario I'm interested in your thoughts around this refactor before I take it any further. 

I appreciate its not directly related to the current PR but the builder itself is a direct result of the current work to make TLS interactions work which is...